runtime/tests/hmac.rs

```diff
@@ -0,0 +1,86 @@
+// Licensed under the Apache-2.0 license.
+pub mod common;
+
+use caliptra_hw_model::HwModel;
+use caliptra_runtime::{CommandId, HmacVerifyCmd};
+use common::run_rom_test;
+use zerocopy::AsBytes;
+
+#[test]
+fn hmac_cmd_run_wycheproof() {
+    let mut model = run_rom_test("mbox");
+
+    model.step_until(|m| {
+        m.soc_mbox().status().read().mbox_fsm_ps().mbox_idle()
+            && m.soc_ifc().cptra_boot_status().read() == 1
+    });
+
+    #[allow(dead_code)]
+    #[derive(Debug)]
+    struct WycheproofResults {
+        id: usize,
+        comment: String,
+    }
+
+    // Collect all the errors and print at the end
+    let mut wyche_fail: Vec<WycheproofResults> = Vec::new();
+    let mut wyche_ran: Vec<WycheproofResults> = Vec::new();
+    let test_set = wycheproof::mac::TestSet::load(wycheproof::mac::TestName::HmacSha384).unwrap();
+
+    for test_groups in test_set.test_groups {
+        for test in &test_groups.tests {
+            // The mailbox is only implemented for keys and tags of 384 bits
+            if test.key.len() != 48 || test.tag.len() != 48 {
+                continue;
+            }
+            wyche_ran.push(WycheproofResults {
+                id: test.tc_id,
+                comment: test.comment.to_string(),
+            });
+            let mut cmd = HmacVerifyCmd {
+                chksum: 0,
+                key: test.key[..].try_into().unwrap(),
+                tag: test.tag[..].try_into().unwrap(),
+                len: u32::try_from(test.msg.len()).unwrap(),
+                msg: [0u8; 256],
+            };
+            cmd.msg[..test.msg.len()].clone_from_slice(test.msg.as_slice());
```

Comment:
Just to confirm, I assume none of the messages are larger than 256 bytes? (Otherwise I the test would probably panic.

Reply:
I confirm